src/main/java/com/google/firebase/iid/FirebaseInstanceId.java

```diff
@@ -0,0 +1,182 @@
+import com.google.common.base.Strings;
+import com.google.common.io.ByteStreams;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.ImplFirebaseTrampolines;
+import com.google.firebase.internal.FirebaseService;
+import com.google.firebase.internal.NonNull;
+import com.google.firebase.internal.TaskToApiFuture;
+import com.google.firebase.tasks.Task;
+
+import java.util.concurrent.Callable;
+
+/**
+ * This class is the entry point for all server-side Firebase Instance ID actions.
+ *
+ * <p>Enables deleting instance IDs associated with Firebase projects.
+ */
+public class FirebaseInstanceId {
+
+  private static final String IID_SERVICE_URL = "https://console.firebase.google.com/v1";
+
+  private final FirebaseApp app;
+  private final HttpRequestFactory requestFactory;
+  private final JsonFactory jsonFactory;
+  private final String projectId;
+
+  private HttpResponseInterceptor interceptor;
+
+  private FirebaseInstanceId(FirebaseApp app) {
+    HttpTransport httpTransport = app.getOptions().getHttpTransport();
+    GoogleCredentials credentials = ImplFirebaseTrampolines.getCredentials(app);
+    this.app = app;
+    this.requestFactory = httpTransport.createRequestFactory(
+        new HttpCredentialsAdapter(credentials));
+    this.jsonFactory = app.getOptions().getJsonFactory();
+    this.projectId = ImplFirebaseTrampolines.getProjectId(app);
+    checkArgument(!Strings.isNullOrEmpty(projectId),
+        "Project ID is required to access instance ID service. Use a service account credential or "
+            + "set the project ID explicitly via FirebaseOptions. Alternatively you can also "
+            + "set the project ID via the GCLOUD_PROJECT environment variable.");
+  }
+
+  /**
+   * Gets the {@link FirebaseInstanceId} instance for the default {@link FirebaseApp}.
+   *
+   * @return The {@link FirebaseInstanceId} instance for the default {@link FirebaseApp}.
+   */
+  public static FirebaseInstanceId getInstance() {
+    return getInstance(FirebaseApp.getInstance());
+  }
+
+  /**
+   * Gets the {@link FirebaseInstanceId} instance for the specified {@link FirebaseApp}.
+   *
+   * @return The {@link FirebaseInstanceId} instance for the specified {@link FirebaseApp}.
+   */
+  public static synchronized FirebaseInstanceId getInstance(FirebaseApp app) {
+    FirebaseInstanceIdService service = ImplFirebaseTrampolines.getService(app, SERVICE_ID,
+        FirebaseInstanceIdService.class);
+    if (service == null) {
+      service = ImplFirebaseTrampolines.addService(app, new FirebaseInstanceIdService(app));
+    }
+    return service.getInstance();
+  }
+
+  @VisibleForTesting
+  void setInterceptor(HttpResponseInterceptor interceptor) {
+    this.interceptor = interceptor;
+  }
+
+  /**
+   * Deletes the specified instance ID from Firebase.
+   *
+   * <p>This can be used to delete an instance ID and associated user data from a Firebase project,
+   * pursuant to the General Data Protection Regulation (GDPR).
```

Comment:
Are you sure that we are ok with includin this legal context in the JS Doc?

Reply:
Good point. I'll check with the IID team before merging this.

Reply:
(I guess the same comment is valid for all repos)